### PR TITLE
planner: support `default_collation_for_utf8mb4` for _utf8mb4'text'

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/46666

Problem Summary:
Support the behavior:

> Any statement containing a string literal of the form _utf8mb4'some text' without a COLLATE clause.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -977,6 +977,9 @@ type SessionVars struct {
 	// SkipUTF8Check check on input value.
 	SkipUTF8Check bool
 
+	// DefaultCollationForUTF8MB4 indicates the default collation of UTF8MB4.
+	DefaultCollationForUTF8MB4 string
```
Comment (zimulala) [inline sessionctx/variable/session.go:981]: We need add this variable because https://github.com/pingcap/tidb/issues/46737

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by AilinKid — COMMENTED** [review]
LGTM

**Review by AilinKid — APPROVED** [review]

**Comment by zimulala** [comment]
Encounter: https://github.com/pingcap/tidb/issues/46841

**Comment by easonn7** [comment]
/approve

There is no sysvar change.

**Pull request merged**